bindings/python/setup.py

```diff
@@ -264,29 +398,24 @@ def build_extension_with_b2(self):
         # write_b2_python_config for limitations on controlling the filename.
 
         # Jamfile hack to copy the module to our target directory
-        args.append(f"python-install-path={target.parent}")
-        args.append("install_module")
+        self._maybe_add_arg(f"python-install-path={target.parent}")
+        self._maybe_add_arg("install_module")
 
         # We use a "project-config.jam" to instantiate a python environment
         # to exactly match the running one.
-        python_binding_dir = pathlib.Path(__file__).parent.absolute()
         config = open(python_binding_dir / 'project-config.jam', 'w+')
-#        config = tempfile.NamedTemporaryFile(mode="w+", delete=False)
         try:
             write_b2_python_config(config)
             config.seek(0)
             log.info("project-config.jam contents:")
             log.info(config.read())
             config.close()
-#            args.append(f"--project-config={config.name}")
+            yield
```

Comment:
This version still unconditionally updates `project-config.jam`. I think this can still collide with the user's intent, if they want to use their own.

Even if the user just wants to use their own `using python : 3.6 : ...` line from their own config, we may override their intent by writing `using python : 3.6 : ...` in our local `project-config.jam`. The `libtorrent-python=on` feature was meant to force this to happen, but it could happen accidentally.

I know it's a little different here because we write to `project-config.jam` instead of adding an arg pointing to a tempfile path. But I think we can check `self._should_add_arg("--project-config")` to gate whether or not we write `project-config.jam`. This way users can do `setup.py build_ext --no-autoconf=--project-config`, if they want to use their own, but we won't actually add an arg if they want it to be autoconfigured.

Reply:
> This version still unconditionally updates project-config.jam. I think this can still collide with the user's intent, if they want to use their own.

Right, but this branch supports an old enough version of boost, before the `--project-config` argument was a thing.

> 
I know it's a little different here because we write to project-config.jam instead of adding an arg pointing to a tempfile path. But I think we can check self._should_add_arg("--project-config") to gate whether or not we write project-config.jam

makes sense